tests/users/test_api_home_statistics.py

```diff
@@ -0,0 +1,246 @@
+from flask import json
+
+from app.database.models.mentorship_relation import MentorshipRelationModel
+from app.database.models.tasks_list import TasksListModel
+from app.database.models.user import UserModel
+from app.database.sqlalchemy_extension import db
+from app.utils.enum_utils import MentorshipRelationState
+from tests.base_test_case import BaseTestCase
+from tests.test_utils import get_test_request_header
+
+
+class TestHomeStatisticsApi(BaseTestCase):
+
+    def setUp(self):
+        super(TestHomeStatisticsApi, self).setUp()
+
+        self.user1 = UserModel("User1", "user1", "password", "test@email.com", True)
+        self.user2 = UserModel("User2", "user2", "password", "test2@email.com", True)
+        self.user1.available_to_mentor = True
+        self.user2.need_mentoring = True
+
+        db.session.add(self.user1)
+        db.session.add(self.user2)
+        db.session.commit()
+
+    def test_relations_non_auth(self):
+        expected_response = {'message': 'The authorization token is missing!'}
+        actual_response = self.client.get('/home', follow_redirects=True)
+        self.assertEqual(401, actual_response.status_code)
+        self.assertEqual(expected_response, json.loads(actual_response.data))
+
+    def test_relations_invalid_id(self):
+        auth_header = get_test_request_header(None)  # Supply invalid user ID for the test
+        actual_response = self.client.get('/home', follow_redirects=True, headers=auth_header)
+        self.assertEqual({'message': 'User not found'}, json.loads(actual_response.data))
+
+    def test_pending_requests_auth(self):
+        start_date = datetime.now()
+
+        end_date = start_date + timedelta(weeks=4)
+        start_date = start_date.timestamp()
+        end_date = end_date.timestamp()
+        tasks_list = TasksListModel()
+
+        mentorship_relation = MentorshipRelationModel(action_user_id=self.user2.id,
+                                                      mentor_user=self.user1,
+                                                      mentee_user=self.user2,
+                                                      creation_date=start_date,
+                                                      end_date=end_date,
+                                                      state=MentorshipRelationState.PENDING,
+                                                      notes="",
+                                                      tasks_list=tasks_list)
+
+        db.session.add(mentorship_relation)
+        db.session.commit()
+        expected_response = {
+            'name': 'User1',
+            'pending_requests': 1,
+            'accepted_requests': 0,
+            'rejected_requests': 0,
+            'completed_relations': 0,
+            'cancelled_relations': 0,
+            'achievements': []
+        }
+        auth_header = get_test_request_header(self.user1.id)
+        actual_response = self.client.get('/home', follow_redirects=True, headers=auth_header)
```

Comment:
Can you also test the status code here?
```
 self.assertEqual(200, actual_response.status_code)
```

And wherever you test a call to the API?

Reply:
Good idea, I've now also added this test to all the API call tests. Thanks.